core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java

```diff
@@ -151,54 +149,47 @@ public String getPath() {
 
   @Override
   public long getSize() {
-    return fileStatus.getLen();
+    try {
+      return fileSystem.getFileStatus(path).getLen();
```

Comment:
Why remove the fileStatus member? Here it will call getFileStatus multiple times

Reply:
fileStatus should not be a member variable because if the same carbonfile object is used to check for exists after the file is deleted by another client then the existing file status would return true instead of false.

There are many such methods in filestatus which can give false information